pkg/graveler/committed/diff.go

```diff
@@ -143,16 +154,20 @@ func (d *diffIterator) compareDiffIterators() diffIteratorCompareResult {
 	rightStartRange := rightRange != nil && d.rightValue.record == nil
 	leftBeforeRight := leftStartRange && rightRange == nil
 	rightBeforeLeft := rightStartRange && leftRange == nil
+	sameBounds := false
 	if leftStartRange && rightStartRange {
 		leftBeforeRight = bytes.Compare(leftRange.MaxKey, rightRange.MinKey) < 0
 		rightBeforeLeft = bytes.Compare(rightRange.MaxKey, leftRange.MinKey) < 0
+		sameBounds = bytes.Equal(leftRange.MinKey, rightRange.MinKey) && bytes.Equal(leftRange.MaxKey, rightRange.MaxKey)
 	}
 	comp := d.compareDiffKeys()
 	switch {
 	case leftBeforeRight:
 		return diffItCompareResultLeftRangeBeforeRight
 	case rightBeforeLeft:
 		return diffItCompareResultRightRangeBeforeLeft
+	case leftStartRange && rightStartRange && comp == 0 && sameBounds:
```

Comment:
```suggestion
	case comp == 0 && sameBounds:
```
Should be enough?

Reply:
In our iterator we know for sure that we are at the beginning of a range if the record is nil. Currently in the case of `sameBounds` we don't receive both range and record ( `sameBounds` => `leftSrartRange` && `rightStartRange`) meaning, your suggestion will work. But may not always be the case, I prefer to leave it as is.